proof/refine/AARCH64/IpcCancel_R.thy

```diff
@@ -1367,6 +1167,32 @@ crunches cancel_ipc
 
 end
 
+crunches asUser
+  for valid_sched_pointers[wp]: valid_sched_pointers
+  (wp: crunch_wps)
+
+crunches set_thread_state
+  for in_correct_ready_q[wp]: in_correct_ready_q
+  (wp: crunch_wps)
+
+crunches set_thread_state_ext
+  for ready_qs_distinct[wp]: ready_qs_distinct
+  (wp: crunch_wps)
+
+lemma set_thread_state_ready_qs_distinct[wp]:
+  "set_thread_state ref ts \<lbrace>ready_qs_distinct\<rbrace>"
+  unfolding set_thread_state_def
+  apply (wpsimp wp: set_object_wp)
+  apply (clarsimp simp: ready_qs_distinct_def)
+  done
+
+lemma as_user_ready_qs_distinct[wp]:
+  "as_user tptr f \<lbrace>ready_qs_distinct\<rbrace>"
+  unfolding as_user_def
+  apply (wpsimp wp: set_object_wp)
+  apply (clarsimp simp: ready_qs_distinct_def)
```

Comment:
I was wondering how we might be able to crunch these, but don't see it, can do the squash into a `by` though (2 applies)

Reply:
I wasn't sure whether `unfolding blah by method1 method2` was too much, but sure, I can do that.

Reply:
We do it a fair bit. Usually `method1` being some `wpsimp` flavour, and `method2` being mop-up.